pulp_smash/tests/rpm/api_v2/test_export.py

```diff
@@ -53,15 +55,205 @@ def _run_getenforce(server_config):
+        })
+        self.change_export_dir_owner(export_dir)
+        return export_dir
+
+    def _publish_to_web(self, distributor):
+        """Publish repository to web using the ``distributor`` argument.
+
+        Return the updated distributor information.
+        """
+        client = api.Client(self.cfg, api.json_handler)
+        path = urljoin(self.repo['_href'], 'actions/publish/')
+        client.post(path, {'id': distributor['id']})
+        return client.get(distributor['_href'])
+
+    def _assert_checksum_type(self, export_dir, distributor, checksum_type):
+        repomd_path = os.path.join(
+            export_dir,
+            distributor['config']['relative_url'],
+            'repodata',
+            'repomd.xml'
+        )
+        document = minidom.parseString(cli.Client(self.cfg).run(
+            'cat {}'.format(repomd_path).split()).stdout)
+        self.assertTrue(all([
+            element.attributes.get('type').value == checksum_type
+            for element in document.getElementsByTagName('checksum')
+        ]))
+
+    def test_publish_to_dir_checksum_type(self):  # pylint:disable=invalid-name
+        """Publish to a directory choosing the checksum type."""
+        for distributor in self.distributors:
+            checksum_type = distributor['config']['checksum_type']
+            with self.subTest(msg=checksum_type):
+                self._assert_checksum_type(
+                    self._publish_to_dir(distributor),
+                    distributor,
+                    checksum_type
+                )
+
+    def test_publish_to_web_checksum_type(self):  # pylint:disable=invalid-name
+        """Publish to web choosing the checksum type."""
+        client = cli.Client(self.cfg)
+        for distributor in self.distributors:
+            checksum_type = distributor['config']['checksum_type']
+            with self.subTest(msg=checksum_type):
+                distributor = self._publish_to_web(distributor)
+                url = _get_iso_url(self.cfg, self.repo, distributor)
+                export_dir = client.run(
+                    'mktemp --directory'.split()).stdout.strip()
+                iso_path = client.run('mktemp'.split()).stdout.strip()
+                self.addCleanup(
+                    client.run,
+                    '{}rm -rf {} {}'.format(
+                        self.sudo(), export_dir, iso_path
+                    ).split()
+                )
+                client.run(
+                    'curl --insecure -o {} {}'.format(iso_path, url).split())
+                client.run((self.sudo() + 'mount -o loop {} {}'.format(
+                    iso_path, export_dir)).split())
+                self._assert_checksum_type(
+                    export_dir,
+                    distributor,
+                    checksum_type
+                )
+                client.run(
+                    (self.sudo() + 'umount {}'.format(export_dir)).split())
```

Comment:
Actually the umount is needed to clean up the loopback device in order to reuse. That said I had to use this way instead of adding as a cleanup. Since now tests are passing I will give it another try and check if it was due to something I did wrong before.